flink-table-store-core/src/test/java/org/apache/flink/table/store/table/SchemaEvolutionTableTestBase.java

```diff
@@ -0,0 +1,301 @@
+                    new DataField(6, "a", new AtomicDataType(DataTypes.INT().getLogicalType())),
+                    new DataField(7, "f", new AtomicDataType(DataTypes.STRING().getLogicalType())),
+                    new DataField(8, "b", new AtomicDataType(DataTypes.STRING().getLogicalType())));
+    protected static final List<String> PARTITION_NAMES = Collections.singletonList("pt");
+    protected static final List<String> PRIMARY_KEY_NAMES = Arrays.asList("pt", "kt");
+
+    protected Path tablePath;
+    protected String commitUser;
+    protected final Configuration tableConfig = new Configuration();
+
+    @TempDir java.nio.file.Path tempDir;
+
+    @BeforeEach
+    public void before() throws Exception {
+        tablePath = new Path(TestAtomicRenameFileSystem.SCHEME + "://" + tempDir.toString());
+        commitUser = UUID.randomUUID().toString();
+        tableConfig.set(CoreOptions.PATH, tablePath.toString());
+        tableConfig.set(CoreOptions.BUCKET, 2);
+    }
+
+    @AfterEach
+    public void after() throws IOException {
+        // assert all connections are closed
+        FileSystem fileSystem = tablePath.getFileSystem();
+        assertThat(fileSystem).isInstanceOf(TraceableFileSystem.class);
+        TraceableFileSystem traceableFileSystem = (TraceableFileSystem) fileSystem;
+
+        java.util.function.Predicate<Path> pathPredicate =
+                path -> path.toString().contains(tempDir.toString());
+        assertThat(traceableFileSystem.openInputStreams(pathPredicate)).isEmpty();
+        assertThat(traceableFileSystem.openOutputStreams(pathPredicate)).isEmpty();
+    }
+
+    protected List<String> getPrimaryKeyNames() {
+        return PRIMARY_KEY_NAMES;
+    }
+
+    protected abstract FileStoreTable createFileStoreTable(Map<Long, TableSchema> tableSchemas);
+
+    public static <R> void writeAndCheckFileResult(
+            Function<Map<Long, TableSchema>, R> firstChecker,
+            BiConsumer<R, Map<Long, TableSchema>> secondChecker,
+            List<String> primaryKeyNames,
+            Configuration tableConfig,
+            Function<Map<Long, TableSchema>, FileStoreTable> createFileStoreTable)
+            throws Exception {
+        Map<Long, TableSchema> tableSchemas = new HashMap<>();
+        tableSchemas.put(
+                0L,
+                new TableSchema(
+                        0,
+                        SCHEMA_0_FIELDS,
+                        5,
+                        PARTITION_NAMES,
+                        primaryKeyNames,
+                        tableConfig.toMap(),
+                        ""));
+        FileStoreTable table = createFileStoreTable.apply(tableSchemas);
+        TableWrite write = table.newWrite("user");
+        TableCommit commit = table.newCommit("user");
+
+        write.write(
+                GenericRowData.of(
+                        StringData.fromString("S001"),
```

Comment:
Maybe we can create a util method, `GenericRowData rowData(Object... values)`, we iterate over each element. If there is a String, it will be converted to StringData.